src/libs/auth.ts

```diff
@@ -109,20 +120,34 @@ const getSigninArgs = (includeBillingScope) => {
 
 export const signIn = async (includeBillingScope = false) => {
   const args = getSigninArgs(includeBillingScope);
-  const user = await getAuthInstance().signinPopup(args);
-
+  const user: User = await getAuthInstance().signinPopup(args);
+  const generatedUuid = uuid();
+  const sessionStartTime = Date.now();
+  authStore.update((state) => ({
+    ...state,
+    sessionId: generatedUuid,
```

Comment:
```suggestion
  const sessionId = uuid();
  const sessionStartTime = Date.now();
  authStore.update((state) => ({
    ...state,
    sessionId,
```

This comment is extremely nit picky. I think naming the variable `sessionId` clarifies what the generated UUID is intended to represent and it also makes the authStore update a little more concise.